x-pack/plugins/ml/server/new_platform/plugin.ts

```diff
@@ -0,0 +1,240 @@
+// @ts-ignore: could not find declaration file for module
+import { jobRoutes } from '../routes/anomaly_detectors';
+// @ts-ignore: could not find declaration file for module
+import { dataFeedRoutes } from '../routes/datafeeds';
+// @ts-ignore: could not find declaration file for module
+import { indicesRoutes } from '../routes/indices';
+// @ts-ignore: could not find declaration file for module
+import { jobValidationRoutes } from '../routes/job_validation';
+// @ts-ignore: could not find declaration file for module
+import { makeMlUsageCollector } from '../lib/ml_telemetry';
+// @ts-ignore: could not find declaration file for module
+import { notificationRoutes } from '../routes/notification_settings';
+// @ts-ignore: could not find declaration file for module
+import { systemRoutes } from '../routes/system';
+// @ts-ignore: could not find declaration file for module
+import { dataFrameRoutes } from '../routes/data_frame';
+// @ts-ignore: could not find declaration file for module
+import { dataRecognizer } from '../routes/modules';
+// @ts-ignore: could not find declaration file for module
+import { dataVisualizerRoutes } from '../routes/data_visualizer';
+// @ts-ignore: could not find declaration file for module
+import { calendars } from '../routes/calendars';
+// @ts-ignore: could not find declaration file for module
+import { fieldsService } from '../routes/fields_service';
+// @ts-ignore: could not find declaration file for module
+import { filtersRoutes } from '../routes/filters';
+// @ts-ignore: could not find declaration file for module
+import { resultsServiceRoutes } from '../routes/results_service';
+// @ts-ignore: could not find declaration file for module
+import { jobServiceRoutes } from '../routes/job_service';
+// @ts-ignore: could not find declaration file for module
+import { jobAuditMessagesRoutes } from '../routes/job_audit_messages';
+// @ts-ignore: could not find declaration file for module
+import { fileDataVisualizerRoutes } from '../routes/file_data_visualizer';
+// @ts-ignore: could not find declaration file for module
+import { initMlServerLog } from '../client/log';
+
+export interface MlHttpServiceSetup extends HttpServiceSetup {
+  route(route: ServerRoute | ServerRoute[]): void;
+}
+
+export interface MlXpackMainPlugin extends XPackMainPlugin {
+  status?: any;
+}
+
+export interface MlCoreSetup {
+  addAppLinksToSampleDataset: () => any;
+  injectUiAppVars: (id: string, callback: () => {}) => any;
+  http: MlHttpServiceSetup;
+  savedObjects: SavedObjectsService;
+  usage: {
+    collectorSet: {
+      makeUsageCollector: any;
+      register: (collector: any) => void;
+    };
+  };
+}
+export interface MlInitializerContext extends PluginInitializerContext {
+  legacyConfig: KibanaConfig;
+  log: Logger;
+}
+export interface PluginsSetup {
+  elasticsearch: ElasticsearchPlugin;
+  xpackMain: MlXpackMainPlugin;
+  security: any;
+  // TODO: this is temporary for `mirrorPluginStatus`
```

Comment:
Just to confirm, is this still a `TODO`?

Reply:
This is a `TODO` in the sense that in the legacy `init` function we were passing the instance of the plugin itself to the `mirrorPluginStatus` function. I wanted to note that that may change as the new platform services become available. It might not - I just wanted to remind myself to check as we move forward.